app/src/test/java/echo/server/EchoServerTest.java

```diff
@@ -67,8 +58,51 @@ void should_sendMessage() throws IOException {
         .thenReturn("hello")
         .thenReturn(null);
 
-    echoServer.start();
+    echoServer.run();
 
     verify(mockSocketIO).sendMessage(mockServerConnection, "hello");
   }
+
+  @Test
+  @DisplayName("should return messages to multiple clients when more than one client sends a message")
+  public void should_sendMultipleClientsMessages() throws IOException, InterruptedException {
+    ExecutorService executorService = Executors.newCachedThreadPool();
+
+    // Start the server in a separate thread
+    executorService.execute(() -> {
+      try {
+        ServerRunner.main(null);
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    });
+
+    // Wait for the server to start
+    Thread.sleep(1000);
+
```

Comment:
This sleep is interesting because it slows down your tests. Is this related to using a "real" client instead of a mock?

Reply:
yeah. I have to wait for the each thread to execute/close or it throws an error because actions start happening out of order. I had to do this on an earlier test that used threading as well.